spec.html

```diff
@@ -21710,6 +21890,34 @@ <h1>Expression Rules</h1>
           1. If this |CallExpression| is _call_, return *true*.
           1. Return *false*.
         </emu-alg>
+        <emu-grammar>
+          OptionalExpression :
+            MemberExpression OptionalChain
+            CallExpression OptionalChain
+            OptionalExpression OptionalChain
+        </emu-grammar>
+        <emu-alg>
+          1. Return HasCallInTailPosition of |OptionalChain| with argument _call_.
+        </emu-alg>
+        <emu-grammar>
+          OptionalChain :
+            `?.` `[` Expression `]`
+            `?.` IdentifierName
+            OptionalChain `[` Expression `]`
+            OptionalChain `.` IdentifierName
+        </emu-grammar>
+        <emu-alg>
+          1. Return *false*.
+        </emu-alg>
+        <emu-grammar>
```

Comment:
A final nitpick: Given that the [ordering of these productions was changed](https://github.com/tc39/ecma262/pull/1646#discussion_r345970102), we should probably put the Arguments case before the Expression and IdentifierName cases here too.

Reply:
(Then again, I guess the Member-Call-Optional ordering isn't being preserved here either...hmm.)